dvc/remote/local.py

```diff
@@ -66,7 +67,7 @@ def work_tree(self):
         # GitTree arbitarily. When repo.tree is GitTree, local cache needs to
         # use its own WorkingTree instance.
         if self.repo:
-            return WorkingTree(self.repo.root_dir)
+            return CleanTree(WorkingTree(self.repo.root_dir))
```

Comment:
Same as before, this works as part of a temporary fix but I'm not sure it makes sense as a real solution.

For local remotes, and DVC local cache we should need to wrap the work tree with CleanTree. With a local remote, it should only contain files which have been pushed to it, and we should not care about filtering/ignoring anything inside the remote. Likewise, for cache, we should not care about filtering anything inside `.dvc/cache`.

The main point of this tree refactor was that local remotes and local cache should only be dealing with paths inside the actual remote or cache. We run into issues when we start mixing remote/cache/repo tree paths and treating them all as just "local filesystem paths".